### ISSUE TITLE
No shortcuts. centos-7, xfce (same failure in gaw3-20200922 too), @a3239fd, gcc-7.3.1, LANG=en_US.UTF-8

### ISSUE THREAD

**Issue by sto6** [StefanSchippers/xschem-gaw#8]
When I build this (eventually**), as well the original gaw3-20200922, both run well but: Shortcuts do not work in both. e.g. ctrl-O for File/Open, Z for zoom-in, etc. (Space & return are special cases: work to invoke add-panel). Mouse works.

By comparison shortcuts work in gtk3 apps from yum: e.g. the apps of gtk3-demo, and gedit.
AFAIK the gtk3-demo maybe does _not_ build menus from XML in a string-literal. e.g. `gtk_builder_add_from_string()`.

Any ideas? Someone elsewhere implied they could log all gtk3 keystrokes or events; any pointers towards how to do that?

```
$ gettext -V
gettext (GNU gettext-runtime) 0.19.8.1
...
$ automake --version
automake (GNU automake) 1.13.4
...
$ ./configure --enable-gawsound=no --prefix=/opt/gaw/20220131 CFLAGS=-g
...
** Configuration summary for gaw3 20200922:

   GLIB library version:             2.56.1
   GTK+ library version:             3.22.30
   Alsa version:
   data directory:                   /opt/gaw/20220131/share/gaw
   locale directory:                 /opt/gaw/20220131/share/locale
   ```

These files were installed: `bin/gaw share/gaw/gaw-gtkrc share/gaw/gaw-gtkrc.css`
Original yielded: `bin/gaw share/gaw/gaw-gtkrc share/gaw/gaw-gtkrc.css share/locale/fr/LC_MESSAGES/gaw3.mo share/applications/gaw.desktop share/pixmaps/gaw.png`

**: The gaw3-20200922 build was straight-forward. xschem-gaw had issues, briefly:
A different error required the fix of issue#7:  `GETTEXT_MACRO_VERSION = 0.19` (and the autoconf-based followup).
Then configure script still had two unreplaced lines that were invalid shell syntax:
```
AM_GNU_GETTEXT(external)
AM_GNU_GETTEXT_VERSION(0.18.1)
```
Commented them out, ran configure, then make & make install both failed at end when recursing into po/ dir, but only after yielding a linked & running gaw executable. po/ dir has Makefile.in but no Makefile.

I speculate that fixing the configure and po/ dir issues while worthwhile won't fix shortcuts, since problem is common to original which doesn't suffer those.

**Comment by StefanSchippers** [comment]
have you tried toggling the 'NumLock' key and/or the 'CapsLock' key?

**Comment by sto6** [comment]
yes, thanks, no difference.
I usually compare side-by-side with another app like gtk3-widget-factory (ctrl-D to toggle dark-mode, ctrl-Q to quit); and plain text fields or terminals.

**Comment by sto6** [comment]
May I ask, what is the oldest (lowest number) GTK+ library version you do or can conveniently test on, and what OSs & window-managers?

Perhaps there's a bug-fix I could back-port into my own build of 3.22.30, but looking thru the library code history it so far eludes me; maybe to do with scope & activation. But knowing some gtk versions which don't have the bug would help narrow the search. For that option, I need the minimal possible upgrade (vs custom compile of all the libraries to get 3.24).

I took https://github.com/mhorauer/GTK-Demo-Examples (guidemo/04_classicmenu.tar.gz) and added another menu using gtk_builder_add_from_string. So the app mixes builder & non-builder menus. Only the builder accelerators do not work for me, and via the mouse the builder menu displays & works (consistent with this gaw issue).

This form: `./classicmenu --gtk-debug  keybindings,actions` (applied to any gtk app, including gaw, gedit, ...) verifies that the gtk-library is 'getting' the keystroke events: "Looking up keycode ...". Then if it works it says "found exact match..."

**Comment by StefanSchippers** [comment]
May be this is all related to a language setting ? what locale are you using? Unfortunately i am not the main gaw developer, i only forked the project to add some xschem interoperation functions. And since then i have always had problems with both GNU autotools and GTK GUI, two things that I personally would like to avoid to deal with in any project.
Can you please try setting locale to English and see if keyboard combinations work?

**Comment by StefanSchippers** [comment]
Closing, no further feedback
